wandb/integration/ultralytics/bbox_utils.py

```diff
@@ -0,0 +1,184 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from ultralytics.engine.results import Results
+from ultralytics.models.yolo.detect import DetectionPredictor
+from ultralytics.yolo.utils import ops
+
+import wandb
+
+
+def scale_bounding_box_to_original_image_shape(
+    box, resized_image_shape, original_image_shape, ratio_pad
+) -> List[int]:
+    """YOLOv8 resizes images during training and the label values are normalized based on this resized shape.
+
+    This function rescales the bounding box labels to the original
+    image shape.
+
+    Reference: https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/utils/callbacks/comet.py#L105
+    """
+    resized_image_height, resized_image_width = resized_image_shape
+    # Convert normalized xywh format predictions to xyxy in resized scale format
+    box = ops.xywhn2xyxy(box, h=resized_image_height, w=resized_image_width)
+    # Scale box predictions from resized image scale back to original image scale
+    box = ops.scale_boxes(resized_image_shape, box, original_image_shape, ratio_pad)
+    # # Convert bounding box format from xyxy to xywh for Comet logging
+    box = ops.xyxy2xywh(box)
+    return box.tolist()
+
+
+def get_ground_truth_bbox_annotations(
+    img_idx, image_path, batch, class_name_map=None
+) -> List[Dict[str, Any]]:
+    indices = batch["batch_idx"] == img_idx
+    bboxes = batch["bboxes"][indices]
+    cls_labels = batch["cls"][indices].squeeze(1).tolist()
+
+    class_name_map_reverse = {v: k for k, v in class_name_map.items()}
+
+    if len(bboxes) == 0:
+        wandb.termwarn(f"Image: {image_path} has no bounding boxes labels")
```

Comment:
Perhaps log this only once by passing the `repeat` kwarg to this method.

Reply:
hmmm... why is it resolved if it is not addressed?